crates/rpc/rpc-eth-types/src/simulate.rs

```diff
@@ -57,6 +58,31 @@ impl ToRpcError for EthSimulateError {
     }
 }
 
+/// An executable transaction that is intended for simulation.
+///
+/// The effect for a layer-2 execution client is that it does not charge L1 cost.
+#[derive(Debug)]
+pub struct SimulateTx<Transaction> {
+    envelope: WithEncoded<Recovered<Transaction>>,
+}
```

Comment:
I believe we can avoid this additional helper type if we also `impl ExecutorTx<> for WithEncoded<Recovered>` then we should be able to pass `WithEncoded::new(default, tx)` directly